### ISSUE TITLE
Error when crawling with permissions problems

### ISSUE THREAD

**Issue by mcamp** [ros/rospack#58]
As suggested by ahendrix I'm creating this issue from answer.ros.org.
There is a problem with autocompletition on folder without enough permissions.
There is a better explanation of this bug here:
http://answers.ros.org/question/224134/error-when-crawling-with-permissions-problems/

Hope I'm posting this issue in the correct place

**Comment by trainman419** [comment]
I'm able to reproduce this by doing:

```
mkdir -p /tmp/test/test
chmod a-r /tmp/test
cd /tmp/test/test
rospack list-names >/dev/null
```

`rospack list-names` is one of the commands that is called from within the roscd tab-completion. I redirect stdout to /dev/null so that the error message (on stderr) is more visible.

**Comment by dirk-thomas** [comment]
Thanks for reporting this. Since this is a not very common use case it is not likely that the maintainers will have time to implement this. Please consider contributing a PR to make rospack work in this scenario.

**Comment by kamalpandey1993** [comment]
rospack find is giving runtime errors on my beaglebone board. Other rospack commands are running fine like rospack depends.
Here is the screenshot

![untitled](https://user-images.githubusercontent.com/18480929/34484604-b611f686-efeb-11e7-8621-44a05a6b5f4b.jpg)

**Comment by dirk-thomas** [comment]
@kamalpandey1993 Your issue seems to be unrelated to this ticket. If you fill a separate ticket please make sure to include enough information, in this case e.g. the content of the file which can't be parsed. Please also don't post screen shots but copy the exact message as text in order to make it searchable.

**Comment by justdoGIT** [comment]
@dirk-thomas apologies for the inconvenience. I've raised a new issue #86 . please have a look

**Comment by dirk-thomas** [comment]
Addressed by #87.
